cloudml-template/examples/babyweight-custom/trainer/model.py

```diff
@@ -234,3 +212,24 @@ def construct_hidden_units():
     print("Hidden units structure: {}".format(hidden_units))
 
     return hidden_units
+
+
+def update_learning_rate():
+    """ Updates learning rate using an exponential decay method
+
+    Returns:
+       float - updated (decayed) learning rate
+    """
+    initial_learning_rate = task.HYPER_PARAMS.learning_rate
+    decay_steps = task.HYPER_PARAMS.num_epochs  # decay after each epoch
+    decay_factor = task.HYPER_PARAMS.learning_rate_decay_factor  # if set to 1, then no decay.
+
+    global_step = tf.train.get_global_step()
+
+    # decayed_learning_rate = learning_rate * decay_rate ^ (global_step / decay_steps)
```

Comment:
I updated decay_steps to be:

   decay_steps = task.HYPER_PARAMS.train_steps

Rather than:

  decay_steps = task.HYPER_PARAMS.num_epochs

As discussed in [docs](https://www.tensorflow.org/versions/r1.4/api_docs/python/tf/train/exponential_decay), using this formula:

decayed_learning_rate = learning_rate * decay_rate ^ (global_step / decay_steps)


The original learning_rate can only reach (learning_rate * decay_rate), when global_steps reaches decay_steps.

Which means, if learning_rate= 0.1, and decay_rate = 0.95, then at the end, learning_rate will be 0.1 * 0.95 = 0.095, and it cannot go below this value.

Reply:
Thanks - I misunderstood what `decay_steps` does.  The name suggests "every how many step will the learning rate be decayed".

Reply:
I would call it `train_steps`, but the parameter is named `decay_steps` in the `tf.train.exponential_decay` [function](https://www.tensorflow.org/api_docs/python/tf/train/exponential_decay)

Reply:
It seems that the reason is `tf.train.exponential_decay` is used differently than the way you do here.  The example in the doc decays the learning rate with a factor of 0.96 for every 100K steps, while the training itself could take, say, 10M steps, with a final learning rate of about `0.016*initial_learning_rate`.  

My understanding is that is the common way of using exponential decay, and the reader come have expected.  Since you are recommending a different approach please consider adding to the comment "If set to 0.5, then..." mentioning this difference.